tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tools;
+
+import joptsimple.OptionException;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.admin.AbstractOptions;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListShareGroupsOptions;
+import org.apache.kafka.clients.admin.ListShareGroupsResult;
+import org.apache.kafka.clients.admin.ShareGroupListing;
+import org.apache.kafka.common.ShareGroupState;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.server.util.CommandLineUtils;
+import org.apache.kafka.tools.consumer.group.ConsumerGroupCommandOptions;
+import org.apache.kafka.tools.reassign.Tuple2;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ShareGroupsCommand {
+
+    public static void main(String[] args) {
+        ConsumerGroupCommandOptions opts = new ConsumerGroupCommandOptions(args);
```

Comment:
Hmmm, though doesn't seems right to have ConsumerGroupCommandOptions in ShareGroupsComman, I do get that most of the checks and options would be same but what about where ShareGroup differs from ConsumerGroup in command line options? How are we going to validate that.
I think you should have ShareGroupCommandOptions which might be similar to ConsumerGroupCommandOptions but with respective changes. We can see how we can abstract the common functionality of groups, if needed.

Reply:
I agree. Just make a `ShareGroupCommandOptions` class copying from the consumer group one, and edit it to match what is currently supported. These commands will not quite match and they'll evolve independently too.

Reply:
Actually, if you use `bin/kafka-share-groups.sh --help --bootstrap-server localhost:9092` you can see the help message applies to consumer groups, rather than share groups :)